### ISSUE TITLE
Upgrade to Electron 7

### ISSUE THREAD

**Issue by AGCaesar** [getferdi/ferdi#170]
Arch Linux just upgraded their electron version to 7.
Ferdi build with my package fails of coarse again.:D I just wanted to know if I need to fix the package or Ferdi does not work with electron 7 at the moment :)

**Comment by vantezzen** [comment]
Ferdi is currently using Electron 6 so you should probably update your package to use that. Are there any specific error messages you are getting?

**Comment by AGCaesar** [comment]
Buliding with electron7 finishes normally, the developer console shows this:
Uncaught Error: Module did not self-register.
    at process.func (electron/js2c/asar.js:138)
    at process.func [as dlopen] (electron/js2c/asar.js:138)
    at Object.Module._extensions..node (internal/modules/cjs/loader.js:922)
    at Object.func (electron/js2c/asar.js:138)
    at Object.func [as .node] (electron/js2c/asar.js:147)
    at Module.load (internal/modules/cjs/loader.js:735)
    at Module._load (internal/modules/cjs/loader.js:648)
    at Module._load (electron/js2c/asar.js:715)
    at Function.Module._load (electron/js2c/asar.js:715)
    at Module.require (internal/modules/cjs/loader.js:775)
    at require (internal/modules/cjs/helpers.js:68)
    at bindings (/usr/lib/ferdi/resources/app.asar/node_modules/bindings/bindings.js:112)
    at Object.<anonymous> (/usr/lib/ferdi/resources/app.asar/node_modules/macos-notification-state/lib/index.js:1)
    at Object.<anonymous> (/usr/lib/ferdi/resources/app.asar/node_modules/macos-notification-state/lib/index.js:81)
    at Module._compile (internal/modules/cjs/loader.js:880)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:892)
    at Module.load (internal/modules/cjs/loader.js:735)
    at Module._load (internal/modules/cjs/loader.js:648)
    at Module._load (electron/js2c/asar.js:715)
    at Function.Module._load (electron/js2c/asar.js:715)
    at Module.require (internal/modules/cjs/loader.js:775)
    at require (internal/modules/cjs/helpers.js:68)
    at Object.<anonymous> (/usr/lib/ferdi/resources/app.asar/node_modules/@meetfranz/electron-notification-state/lib/index.js:5)
    at Object.<anonymous> (/usr/lib/ferdi/resources/app.asar/node_modules/@meetfranz/electron-notification-state/lib/index.js:27)
    at Module._compile (internal/modules/cjs/loader.js:880)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:892)
    at Module.load (internal/modules/cjs/loader.js:735)
    at Module._load (internal/modules/cjs/loader.js:648)
    at Module._load (electron/js2c/asar.js:715)
    at Function.Module._load (electron/js2c/asar.js:715)
    at Module.require (internal/modules/cjs/loader.js:775)
    at require (internal/modules/cjs/helpers.js:68)
    at Object.<anonymous> (/usr/lib/ferdi/resources/app.asar/stores/AppStore.js:1)
    at Object.<anonymous> (/usr/lib/ferdi/resources/app.asar/stores/AppStore.js:2)
    at Module._compile (internal/modules/cjs/loader.js:880)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:892)
    at Module.load (internal/modules/cjs/loader.js:735)
    at Module._load (internal/modules/cjs/loader.js:648)
    at Module._load (electron/js2c/asar.js:715)
    at Function.Module._load (electron/js2c/asar.js:715)
    at Module.require (internal/modules/cjs/loader.js:775)
    at require (internal/modules/cjs/helpers.js:68)
    at Object.<anonymous> (/usr/lib/ferdi/resources/app.asar/stores/index.js:1)
    at Object.<anonymous> (/usr/lib/ferdi/resources/app.asar/stores/index.js:2)
    at Module._compile (internal/modules/cjs/loader.js:880)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:892)
    at Module.load (internal/modules/cjs/loader.js:735)
    at Module._load (internal/modules/cjs/loader.js:648)
    at Module._load (electron/js2c/asar.js:715)
    at Function.Module._load (electron/js2c/asar.js:715)

Also the console dispalys "(electron) 'getName function' is deprecated and will be removed. Please use 'name property' instead."
But this is not an acute problem...
I would change the packagebuild to electron6 but Arch does not provide a version....

**Comment by vraravam** [comment]
Is this something that can be bumped up in priority? Older versions of electron have exploits that Imo should be fixed by upgrading electron.

**Comment by vantezzen** [comment]
Ferdi - Just Like Franz - is currently using Electron 6, which is still withing ElectronJS's [support period](https://electronjs.org/docs/tutorial/support#supported-versions), so there shouldn't be any exploits present that Electron 7 doesn't have.
I think we will wait until Franz is upgrading to Electron 7 - but PRs for this are still welcome.

**Comment by kytwb** [comment]
As far as I can see, Electron 7 doesn't have any [breaking change](https://electronjs.org/blog/electron-7-0#breaking-changes) that affects us.

> The file electron.asar no longer exists. Any packaging scripts that depend on its existence should be updated. #18577

@AGCaesar does the above prevent you from packaging the AUR archive?

I upgraded Electron from 6 to 7 on `develop` for now. I also updated all the dependencies with minor updates that were visibly not breaking anything. That includes the latest React 16.x.  
Overall all I noticed is more deprecation warnings, but everything working as usual.
